src/controllers/main/main.ts

```diff
@@ -951,18 +954,15 @@ export class MainController extends EventEmitter {
       if (!network) {
         throw ethErrors.provider.chainDisconnected('Transaction failed - unknown network')
       }
-
       userRequest = {
         id: new Date().getTime(),
         action: {
           kind,
-          calls: [
-            {
-              to: transaction.to,
-              value: transaction.value ? getBigInt(transaction.value) : 0n,
-              data: transaction.data || '0x'
-            }
-          ]
+          calls: calls.map((call) => ({
```

Comment:
i suggest implementing a helper for this conversion that will be used when we want to implement deploying smart contracts

Reply:
A similar logic has already started in https://github.com/AmbireTech/ambire-common/pull/935/files so I'll leave it for there